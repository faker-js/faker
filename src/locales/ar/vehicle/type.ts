export default [
  'تمديد الكابينة بيك آب',
  'سيارة الركاب',
  'سيارة رياضية',
  'سيدان',
  'طاقم الكابينة بيك آب',
  'عربة',
  'كارغو فان',
  'كوبيه',
  'مكشوفة',
  'ميني فان',
  'هاتشباك',
];
