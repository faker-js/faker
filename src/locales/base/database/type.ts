export default [
  'bigint',
  'binary',
  'bit',
  'blob',
  'boolean',
  'date',
  'datetime',
  'decimal',
  'double',
  'enum',
  'float',
  'geometry',
  'int',
  'mediumint',
  'point',
  'real',
  'serial',
  'set',
  'smallint',
  'text',
  'time',
  'timestamp',
  'tinyint',
  'varchar',
];
