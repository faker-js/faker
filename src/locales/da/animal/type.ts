export default [
  'hund',
  'kat',
  'slange',
  'bjørn',
  'løve',
  'hval',
  'insekt',
  'krokodille',
  'ko',
  'fugl',
  'fisk',
  'gnaver',
  'hest',
];
