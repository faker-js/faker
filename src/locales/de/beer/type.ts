export default [
  'Amber Ale',
  'American Pale Ale - APA',
  'Belgian Strong Ale',
  'Bitter',
  'Blonde - Golden Ale',
  'India Pale Ale - IPA',
  'Italian Grape Ale - IGA',
  'Pilsner - Lager',
  'Quadrupel',
  'Saison',
  'Scottish - Scotch Ale',
  'Session IPA',
  'Spicy',
  'Tripel',
  'Berliner Weisse',
  'Blanche/Weiss',
  'Gose',
  'Pale Ale',
  'Bock',
  'Fruit',
  'Marzen',
  'Sour',
  'Stout',
];
