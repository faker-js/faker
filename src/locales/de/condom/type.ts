export default [
  'Billig',
  'Ultra dünn',
  'Dick',
  'Mit ...-Geschmack',
  'Farbig',
  'Strukturiert',
  'Im Dunkeln leuchten',
  'Kribbeln',
  'Erwärmen',
  'Höhepunktverzögerung',
  'Anregend',
  'Latexfrei',
  'Fairer Handel',
  'vegan',
  'Weiblich',
  'Nicht geschmiert',
  'Extra geschmiert',
];
