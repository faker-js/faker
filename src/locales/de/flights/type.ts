export default ['Direktfl√ºge', 'Stop', 'Anschlussflug'];
