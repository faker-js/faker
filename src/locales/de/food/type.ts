export default ['Fisch', 'Fleisch', 'Gem√ºse', 'Obst', 'Brot', 'Milch'];
