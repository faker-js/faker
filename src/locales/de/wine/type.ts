export default ['Weißweine', 'Rotweine', 'Dessertweine', 'Cahmpagne', 'Rosè'];
