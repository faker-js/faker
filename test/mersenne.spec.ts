import { beforeAll, beforeEach, describe, expect, it } from 'vitest';
import type { Mersenne } from '../src/internal/mersenne/mersenne';
import mersenneFn from '../src/internal/mersenne/mersenne';
import { seededRuns } from './support/seededRuns';

const minMaxTestCases = [
  { min: 0, max: 100 },
  { min: -60, max: 0 },
  { min: -50, max: 60 },
];

const NON_SEEDED_BASED_RUN = 25;

describe('mersenne twister', () => {
  let mersenne: Mersenne;

  beforeEach(() => {
    mersenne = mersenneFn();
  });

  for (const seed of [...seededRuns, [42, 1, 2], [1337, 1, 2], [1211, 1, 2]]) {
    describe(`seed: ${JSON.stringify(seed)}`, () => {
      beforeEach(() => {
        mersenne.seed(seed);
      });

      for (const { min, max } of minMaxTestCases) {
        it(`should return deterministic values for next({ min: ${min}, max: ${max} })`, () => {
          const actual = mersenne.next({ min, max });

          expect(actual).toMatchSnapshot();
        });
      }

      it.todo(`should return 0 for next({ min: ${0}, max: ${1} })`, () => {
        const actual = mersenne.next({ min: 0, max: 1 });

        expect(actual).toEqual(0);
      });
    });
  }

  // Create and log-back the seed for debug purposes
  const seeds = [
    Math.ceil(Math.random() * 1_000_000_000),
    [
      Math.ceil(Math.random() * 1_000_000_000),
      Math.ceil(Math.random() * 1_000_000_000),
    ],
  ];

  for (const seed of seeds) {
    describe(`random seeded tests ${JSON.stringify(seed)}`, () => {
      beforeAll(() => {
        mersenne.seed(seed);
      });

      for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
        describe('next', () => {
          it('should return random number from interval [min, max)', () => {
            const actual = mersenne.next({ min: 0, max: 2 });

            expect(actual).toBeGreaterThanOrEqual(0);
            expect(actual).toBeLessThan(2);
          });
        });
      }
    });
  }
});
