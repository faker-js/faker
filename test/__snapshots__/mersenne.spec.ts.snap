// Vitest Snapshot v1

exports[`mersenne twister > seed: [42,1,2] > should return deterministic values for next({ min: -50, max: 60 }) 1`] = `44`;

exports[`mersenne twister > seed: [42,1,2] > should return deterministic values for next({ min: -60, max: 0 }) 1`] = `-9`;

exports[`mersenne twister > seed: [42,1,2] > should return deterministic values for next({ min: 0, max: 100 }) 1`] = `85`;

exports[`mersenne twister > seed: [1211,1,2] > should return deterministic values for next({ min: -50, max: 60 }) 1`] = `48`;

exports[`mersenne twister > seed: [1211,1,2] > should return deterministic values for next({ min: -60, max: 0 }) 1`] = `-7`;

exports[`mersenne twister > seed: [1211,1,2] > should return deterministic values for next({ min: 0, max: 100 }) 1`] = `89`;

exports[`mersenne twister > seed: [1337,1,2] > should return deterministic values for next({ min: -50, max: 60 }) 1`] = `-31`;

exports[`mersenne twister > seed: [1337,1,2] > should return deterministic values for next({ min: -60, max: 0 }) 1`] = `-50`;

exports[`mersenne twister > seed: [1337,1,2] > should return deterministic values for next({ min: 0, max: 100 }) 1`] = `17`;

exports[`mersenne twister > seed: 42 > should return deterministic values for next({ min: -50, max: 60 }) 1`] = `-9`;

exports[`mersenne twister > seed: 42 > should return deterministic values for next({ min: -60, max: 0 }) 1`] = `-38`;

exports[`mersenne twister > seed: 42 > should return deterministic values for next({ min: 0, max: 100 }) 1`] = `37`;

exports[`mersenne twister > seed: 1211 > should return deterministic values for next({ min: -50, max: 60 }) 1`] = `52`;

exports[`mersenne twister > seed: 1211 > should return deterministic values for next({ min: -60, max: 0 }) 1`] = `-5`;

exports[`mersenne twister > seed: 1211 > should return deterministic values for next({ min: 0, max: 100 }) 1`] = `92`;

exports[`mersenne twister > seed: 1337 > should return deterministic values for next({ min: -50, max: 60 }) 1`] = `-22`;

exports[`mersenne twister > seed: 1337 > should return deterministic values for next({ min: -60, max: 0 }) 1`] = `-45`;

exports[`mersenne twister > seed: 1337 > should return deterministic values for next({ min: 0, max: 100 }) 1`] = `26`;
